nixneovim.nix

```diff
@@ -178,6 +180,30 @@ in
           };
         '';
       };
+
+      augroups = mkOption {
+        default = { };
+        type = let inherit (augroups) augroupOptions; in types.attrsOf augroupOptions;
```

Comment:
Why did you make this so complicated and long?

Could you change it to `types.attrsOf augroups.augroupsOptions`?

Reply:
First time actually writing so much nix code, I'm busy this weekend so you can expect this to be cleaned up and done by next weekend